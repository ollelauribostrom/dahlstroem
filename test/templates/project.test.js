import React from 'react';
import { shallow } from 'enzyme';

import ProjectTemplate from '../../src/templates/project';

describe('Tests for Project template', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ProjectTemplate
        location={{ pathname: '/test' }}
        pageContext={{ seo: { title: 'test', keywords: ['a', 'b'] } }}
        data={{
          datoCmsWork: {
            title: 'Test project',
            description: 'test description',
            video1: 'videosrc',
            images: [
              {
                fluid: {
                  aspectRatio: 1.4678899082568808,
                  sizes: '(max-width: 1800px) 100vw, 1800px',
                  src: 'someurl',
                  srcSet: 'someurl',
                },
              },
              {
                fluid: {
                  aspectRatio: 1.4678899082568808,
                  sizes: '(max-width: 1800px) 100vw, 1800px',
                  src: 'someurl',
                  srcSet: 'someurl',
                },
              },
            ],
          },
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
