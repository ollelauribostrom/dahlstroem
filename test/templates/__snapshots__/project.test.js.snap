// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Project template renders correctly 1`] = `
<Layout
  color={
    Object {
      "hex": "#000",
    }
  }
  menuBackground={
    Object {
      "hex": "#000",
    }
  }
  menuColor={
    Object {
      "hex": "#fff",
    }
  }
  path="/test"
>
  <SEO
    keywords={
      Array [
        "a",
        "b",
      ]
    }
    lang="en"
    meta={Array []}
    title="test"
  />
  <h3>
    Test project
  </h3>
  <p>
    test description
  </p>
  <Image
    Tag="div"
    alt=""
    critical={false}
    fadeIn={true}
    fluid={
      Object {
        "aspectRatio": 1.4678899082568808,
        "sizes": "(max-width: 1800px) 100vw, 1800px",
        "src": "someurl",
        "srcSet": "someurl",
      }
    }
    key="someurl"
    style={
      Object {
        "marginBottom": "20px",
      }
    }
  />
  <Image
    Tag="div"
    alt=""
    critical={false}
    fadeIn={true}
    fluid={
      Object {
        "aspectRatio": 1.4678899082568808,
        "sizes": "(max-width: 1800px) 100vw, 1800px",
        "src": "someurl",
        "srcSet": "someurl",
      }
    }
    key="someurl"
    style={
      Object {
        "marginBottom": "20px",
      }
    }
  />
</Layout>
`;
